### PR TITLE
Blog-App: Processing data

### PR DESCRIPTION
**Pull request by kelomo2502**
In this milestone, I did the following

- Create models
- Use models to insert data
- Create custom methods for:
-  Users
     A method that returns the 3 most recent posts for a given user.
- Posts
    A method that updates the posts counter for a user.
    A method which returns the 5 most recent comments for a given post.

- Comments
    A method that updates the comments counter for a post.

- Likes
    A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  after_save :update_comments_counter
+
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_comments_counter
+    post.increment!(:comments_counter)
+  end
```
Comment (rloterh) [inline app/models/comment.rb:9]: - [ ] OPTIONAL
Following best practices `update_comments_counter` should be marked as private after implementing the `after_save` method.

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  after_save :update_likes_counter
+
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_likes_counter
+    post.increment!(:likes_counter)
+  end
```
Comment (rloterh) [inline app/models/like.rb:9]: - [ ] OPTIONAL
Following best practices `update_likes_counter` should be marked as private after implementing the `after_save` method.

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  after_save :update_post_counter
+
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_post_counter
```
Comment (rloterh) [inline app/models/post.rb:8]: - [ ] OPTIONAL
Following best practices `update_post_counter` should be marked as private after implementing the `after_save` method.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Gbenga,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉🌟 

## Highlights 👍
- Model associations properly implemented

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly consider my comments marked **[OPTIONAL]** under this review_

**You can also consider:**
- MIT license link in your ReadMe is broken, kindly endeavor to fix it.
Implementing the above change(s) will make your ReadMe appear more professional.
<br>
Keep soaring higher! 💻 🍷 🚀 
<br><br>
Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 3 above)
Comment (kelomo2502) [reply app/models/post.rb:8]: Noted...Thank you

**Pull request merged**